
##################################################################
# Author: Xin Wang                                                   
# Email: xin.wang@nationwidechildrens.org                         
# Copyright (c) 2025 Kidney and Urology Tract Center
# Nationwide Children's Hospital
# Description: 
#   This script is use to process the intensity matrix:
#   The inputs:
#     1. the output was generated by fragpipe -- combined_protein.tsv, we renamed the sample ID based on the rank of eGFR.
#     2. THe preprocessing of these intensity matrix includes:
#         2.1 remove decoy and contaminants
#         2.2 replace 0 with NA if no NA inlcuded as missing value
#         2.3 filter our proteins with only one non-NA value in a condition
#         2.4 log2 transform the intenisity matrix -- we will ignore the normalization 
#         2.5 Conduct imputation with selected imputation method, "Knn", ""bpca 
#   
################################################################
#BiocManager::install("MSnbase")
library(MSnbase)
#library(diann)
# BiocManager::install("GMSimpute")
#BiocManager::install("SeqKnn")
# BiocManager::install("rrcovNA")
# BiocManager::install("NormalyzerDE")
# BiocManager::install("limma")
#BiocManager::install("DEqMS")
library(GMSimpute)
#library(SeqKnn)
library(rrcovNA)
library(NormalyzerDE)
library(limma)
library(DEqMS)
library(tidyverse)
library(ggplot2)
library(ggnewscale)
library(ggrepel)
#BiocManager::install("pRoloc")
library("pRoloc")
library(edgeR)
library(clusterProfiler)
library(org.Mm.eg.db)
#keytypes(org.Mm.eg.db) 

library(enrichplot)
library()
library(tidyverse)
library(factoextra)
#BiocManager::install("org.Hs.eg.db",force = TRUE)

library(org.Hs.eg.db)
### analyses the proteomic across patients

## read the table

setwd("/Users/XXW004/Documents/Projects/DarylMcleod/Results/Results_03242025/Intensity_Analyses/")

## set up the input dir

InputDir <- c("/Users/XXW004/Documents/Projects/DarylMcleod/RawData/Final_Used/")

## read the table the table containing the sample names, 
## unique protein counts and MaxLFQ Intensity were used for analyses


## preprocessing the datasets: 
#     read the file, 
#     retain the proteins that 0.8 of total samples in each condition
#     normalization
#     imputation


# Read the original datasets and clinical datasets: 
# Note: we have transfer tsv into csv.

pro_file<-read.csv(paste0(InputDir,"combined_final_protein_ByJF.csv"), header = T)
head(pro_file)
length(colnames(pro_file))
intens<-pro_file[, 175:length(colnames(pro_file))]

intens %>% head()
intens[intens==0]<-NA
row.names(intens) <-pro_file[,1]
intens %>% nrow()
# we remove the string .MaxLFQ.Intensity from the colnames
colnames(intens)<-gsub('.MaxLFQ.Intensity','',colnames(intens))

# we reorder the 
Desired_NameOrder <- c("Case1","Case2","Case3","Case4","Case5","Case6","Case7","Case8","Case9","Case10","Case11","Case12","Case13","Case14","Case15","Case16","Case17","Case18","Case19","Case20","Control1","Control2","Control3","Control4","Control5","Control6","Control7","Control8","Control9","Control10","Control11","Control12","Control13","Control14","Control15","Control16","Control17","Control18","Control19","Control20")
# we filtered the samples ~0.9
idx_retain<-which(apply(intens,1,function(x) sum(is.na(x)))<length(intens[1,])*0.9)
intens=intens[idx_retain,Desired_NameOrder]

# check how many of the proteins after the filtering
intens %>% nrow()

# check the filter whether contain UPK2
# the requested protein
RequestedProteins<- "sp|O00526|UPK2_HUMAN"
intens[RequestedProteins,]

# Here we consider to use the log2 to transfer the dataset
intens_tran<-log2(as.matrix(intens))
intens_tran[RequestedProteins,]
table(is.na(intens_tran))
# check the nuber of NAs per protein
## number of NAs per protein
data(naset)

# Processing quantitative data using the impute method: 
# A set of imputation methods are available in the impute method: it takes an MSnSet instance as input, the name of the imputation method to be applied (one of bpca, knn, QRILC, MLE, MLE2, MinDet, MinProb, min, zero, mixed, nbavg, with, RF, none), possible additional parameters and returns an updated for MSnSet without any missing values. Below, we apply a deterministic minimum value imputation on the naset example data:

# get the protein names:
fd <- data.frame(row.names(intens))
row.names(fd)<-row.names(intens)
# get the sample names:
pd <- data.frame(colnames(intens))
row.names(pd) = colnames(intens)
intens_imp<-MSnSet(intens_tran, fd, pd)

table(is.na(intens_imp))
# as the sample have already been normalized, we did not perform the step

# We perfomed the imputation using Knn

intens_imp<-MSnbase::impute(intens_imp, "knn")

table(is.na(intens_imp))

# check the expression of these samples
exprs(intens_imp) # we have already log tranfer the dataset

# we then use the quanitfile normalization
intens_imp.vsn <- normalise(intens_imp, "vsn")

exprs(intens_imp.vsn) 
# then plot the PCA results
##### generate a PCA analysese

colorDiff<-c(rep("red",20), rep("blue",20))

mds <- plotMDS(exprs(intens_imp), col=colorDiff)
toplot <- data.frame(Dim1 = mds$x, Dim2 = mds$y)
library(ggplot2)
#ggplot(toplot, aes(Dim1, Dim2))+ geom_point() +
# geom_text_repel(label=label_name,segment.size=0.01,hjust=0.2, size=3,  col=c(rep("gray",5),rep("black",4),rep("blue",4), rep("red",4), rep("yellow",4),rep("violet",4),rep("brown",4),rep("blue",4),))+
#  theme_classic()
#geom_text( label = label_name, nudge_x = 0.1, nudge_y = 0.1,  check_overlap = T)+
#theme_bw() + 
#  theme(panel.border = element_blank(), panel.grid.major = element_blank(),panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"))
#Gcol<-c(rep("gray",5),rep("black",4),rep("blue",4), rep("red",4),rep("brown",4),rep("yellow",4),rep("navy blue",3), rep("purple",4),rep("LimeGreen",4),rep("pink",4),rep("orange",4), rep("magenta",4))
pdf("PCAanalyses.pdf", width = 6, height = 6)
ggplot(toplot, aes(Dim1, Dim2))+ geom_point(col=colorDiff) +
  geom_text_repel(label=colnames(exprs(intens_imp)),segment.size=0.01,hjust=0.2, size=3, col=colorDiff,max.overlaps=20)+
  theme_classic()+
  #geom_text( label = label_name, nudge_x = 0.1, nudge_y = 0.1,  check_overlap = T)+
  theme_bw() + 
  theme(panel.border = element_blank(), panel.grid.major = element_blank(),panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"))

dev.off()
### 

# then save as the Proprocessing objects

# save the imputation and noramlized 
saveRDS(intens_imp.vsn, "PUV_knnimputation_vsnnormalization_intens_imp.vsn.RDS")

# 

# save the imputation but without normalization
saveRDS(intens_imp, "PUV_knnimputation_withoutnormalization_intens_imp.RDS")



# 
# 
# 
# 
# 
# ### pay attention we change the file into csv file 
# ProteomicData<- read.csv(file="Proteomics_AcrossPatients_Updates2.csv",header = T)
# 
# ClinicalFactors<-read.csv(file = "ClinalFactor_FromLindsey2.csv", row.names  = 1, header=T, sep = ",")
# 
# ClinicalFactors 
# 
# ## measure the basic statistics of clinical factors
# summary(ClinicalFactors)
# sd(ClinicalFactors$age)
# mean(ClinicalFactors$age)
# quantile(ClinicalFactors$age, probs =c(0.05,0.95))
# quantile(ClinicalFactors$Weight, probs =c(0.05,0.95))
# quantile(ClinicalFactors$Height, probs =c(0.05,0.95))
# quantile(ClinicalFactors$Creatinin, probs =c(0.05,0.95))
# quantile(ClinicalFactors$GFR, probs =c(0.05,0.95))
# #apply(ClinicalFactors, 2, function(x) quantile(x, probs =c(0.05,.95),na.rm = FALSE) )
# 
# ClinicalFactorsControl<-read.csv(file = "ControlClinic2.csv", row.names  = 1, header=T, sep = ",")
# quantile(ClinicalFactorsControl$age, probs =c(0.05,0.95))
# summary(ClinicalFactorsControl)
# ClinicalFactors2<-as.matrix(t(ClinicalFactors))
# ClinicalFactors2
# ## perform the T test for the groups
# 
# ProteomicMatrix<-ProteomicData[,c(8:47)]
# 
# head(ProteomicMatrix)
# ### merge with the clinical Factor
# 
# head(ProteomicMatrix)
# head(ClinicalFactors)
# ColumnExpressedGenes<-apply(ProteomicMatrix,2, function(x) sum(x>0))
# 
# ColumnExpressedGenes<-as.matrix(ColumnExpressedGenes)
# ColumnExpressedGenes
# #rownames(ColumnExpressedGenes)
# 
# #ColumnExpressedGenes$ID= rownames(ColumnExpressedGenes)
# #colnames(ColumnExpressedGenes)<-c("ExpressedProtein")
# 
# nrow(ColumnExpressedGenes)
# 
# #intersect(ColumnExpressedGenes, ClinicalFactors, by )
# 
# rownames(ClinicalFactors2)
# rownames(ColumnExpressedGenes)
# 
# ### merge two matrix by their row names
# 
# ClinicalFactors
# ColumnExpressedGenes
# 
# 
# AllProteomicClinic<- merge(x=ColumnExpressedGenes, y=ClinicalFactors, by="row.names", all.y=TRUE, all.x=TRUE)
# 
# AllProteomicClinic
# # average of protoemics
# AllProteomicClinic %>% mutate(Group = case_when(!is.na(age) ~ "Case", TRUE ~ "Control" )) %>% group_by(Group) %>% summarise(aveg= mean(V1))
# AllProteomicClinic %>% mutate(Group = case_when(!is.na(age) ~ "Case", TRUE ~ "Control" )) %>% group_by(Group) %>% summarise(quantile0.05=quantile(V1, probs =0.05), aveg= mean(V1),quantile0.95=quantile(V1, probs =0.95), maxium=max(V1), medianvalue =median(V1) )
# 
# AllProteomicClinic <- AllProteomicClinic %>% mutate(Group = case_when(!is.na(age) ~ 1, TRUE ~ 0))
# write.csv(ColumnExpressedGenes, file="ExpressedGenesNumberAcrossSamples.csv")
# ### give the row name
# rownames(ProteomicMatrix)<-ProteomicData[,3]
# 
# row.names(ProteomicMatrix)
# #summary(ProteomicMatrix)
# 
# ## check the cellular components of these proteins
# CellularComponentsTotalProteinsgroup<-groupGO(gene= rownames(ProteomicMatrix), OrgDb = "org.Hs.eg.db", ont='CC',keyType = 'SYMBOL',level = 3, readable = TRUE)
# 
# CellularComponentsTotalProteinsEnriche<-enrichGO(rownames(ProteomicMatrix), OrgDb = "org.Hs.eg.db", ont='CC',pAdjustMethod = 'BH',pvalueCutoff = 1, qvalueCutoff = 1,keyType = 'SYMBOL')
# 
# # save the enriched protein
# write.csv(CellularComponentsTotalProteinsEnriche, file = "CellularComponentsTotalProteinsEnriche2.csv")
# 
# ## draw the figures
# pdf("TotalProtein_CellularComponentsEnrichments2.pdf", width = 8, height = 5)
# 
# barplot(CellularComponentsTotalProteinsEnriche, showCategory = 15,font.size = 8,label_format = 80) + coord_flip() + theme(axis.text.x = element_text(angle = 45, hjust = 1)) 
# 
# dev.off()
# #CellularComponentsTotalProteinsgroup@result %>% dplyr::arrange(desc(Count),.by_group = TRUE) %>% dplyr::select(Description, Count, GeneRatio)
# ## check the overlapping proteins between cases and controls
# #ProteomicMatrix %>% mutate(GeneCluste= case_when(sum()))
# ?arrange
# #expressed protein in the case
# ExpressedCase<-rowSums(ProteomicMatrix[1:20]) >0
# # 
# ExpressedCaseGenes<-rownames(ProteomicMatrix[ExpressedCase,])
# 
# ExpressedControl<-rowSums(ProteomicMatrix[21:40]) >0
# # 
# ExpressedCControlGenes<-rownames(ProteomicMatrix[ExpressedControl,])
# 
# length(intersect(ExpressedCaseGenes,ExpressedCControlGenes))
# 
# length(ExpressedCControlGenes)
# 
# length(ExpressedCaseGenes)
# 
# 
# ## check the GO enrichment of cellular function
# AllProteinCC<-enrichGO(row.names(ProteomicMatrix), OrgDb = "org.Hs.eg.db", ont='CC',pAdjustMethod = 'BH',pvalueCutoff = 1, qvalueCutoff = 1,keyType = 'SYMBOL')
# AllProteinCCSimplify<-simplify(AllProteinCC)
# 
# barplot(AllProteinCCSimplify, showCategory = 20,font.size = 12,label_format = 80)
# 
# AllProteinCCGroup<-groupGO(row.names(ProteomicMatrix), OrgDb = "org.Hs.eg.db", ont='CC', level = 3,keyType = 'SYMBOL', readable = TRUE)
# 
# 
# #CgoDown<-enrichGO(row.names(Downregulated), OrgDb = "org.Hs.eg.db", ont='BP',pAdjustMethod = 'BH',pvalueCutoff = 1, qvalueCutoff = 1,keyType = 'SYMBOL')
# 
# ## filter out the matrix that keep<- rowSums(y$counts) >5
# ## total counts less than 5
# keep<-rowSums(ProteomicMatrix) >=5
# ## samples zero sample should less than 20 
# 
# genelist.filted<-ProteomicMatrix[keep, ]
# ## how many final genes were used for final results
# dim(genelist.filted)
# #genelist.filted
# rowSums(genelist.filted==0)
# ### check how many sample with zero in each 
# genelist.filted<-genelist.filted[rowSums(genelist.filted==0)<=35,]
# dim(genelist.filted)
# rowSums(genelist.filted==0)
# 
# 
# ### We used quantile normalization for the 
# genelist.filted<-normalizeQuantiles(genelist.filted)
# 
# ### write into a file
# write.csv(genelist.filted, file = "Normalized_genelist.filted2.csv")
# ##### generate a PCA analysese
# 
# colorDiff<-c(rep("red",20), rep("blue",20))
# 
# mds <- plotMDS(genelist.filted, col=colorDiff)
# toplot <- data.frame(Dim1 = mds$x, Dim2 = mds$y)
# library(ggplot2)
# #ggplot(toplot, aes(Dim1, Dim2))+ geom_point() +
# # geom_text_repel(label=label_name,segment.size=0.01,hjust=0.2, size=3,  col=c(rep("gray",5),rep("black",4),rep("blue",4), rep("red",4), rep("yellow",4),rep("violet",4),rep("brown",4),rep("blue",4),))+
# #  theme_classic()
# #geom_text( label = label_name, nudge_x = 0.1, nudge_y = 0.1,  check_overlap = T)+
# #theme_bw() + 
# #  theme(panel.border = element_blank(), panel.grid.major = element_blank(),panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"))
# #Gcol<-c(rep("gray",5),rep("black",4),rep("blue",4), rep("red",4),rep("brown",4),rep("yellow",4),rep("navy blue",3), rep("purple",4),rep("LimeGreen",4),rep("pink",4),rep("orange",4), rep("magenta",4))
# pdf("PCAanalyses2.pdf", width = 6, height = 6)
# ggplot(toplot, aes(Dim1, Dim2))+ geom_point(col=colorDiff) +
#   geom_text_repel(label=colnames(genelist.filted),segment.size=0.01,hjust=0.2, size=3, col=colorDiff,max.overlaps=20)+
#   theme_classic()+
#   #geom_text( label = label_name, nudge_x = 0.1, nudge_y = 0.1,  check_overlap = T)+
#   theme_bw() + 
#   theme(panel.border = element_blank(), panel.grid.major = element_blank(),panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"))
# 
# dev.off()
# ### 
# 
# # other methods: Kmeans
# #install.packages("factoextra")
# 
# df<-scale(t(genelist.filted))
# Km<-kmeans(df, centers = 5, nstart=25)
# 
# fviz_cluster(Km, data = df, ellipse.type = "convex", palette = "jco", ggtheme = theme_minimal())
# 
# fviz_nbclust(nor, kmeans, method = "wss")
# ### other methods: 
# 
# 
# #######################################################
# # only for patients
# #######################################################
# #### we choose only the patients for the clustering
# 
# Patients<- genelist.filted[,c(1:20)]
# 
# ## quantile normalization
# #Patients<-normalizeQuantiles(Patients)
# 
# mds <- plotMDS(Patients)
# toplot <- data.frame(Dim1 = mds$x, Dim2 = mds$y)
# library(ggplot2)
# #ggplot(toplot, aes(Dim1, Dim2))+ geom_point() +
# # geom_text_repel(label=label_name,segment.size=0.01,hjust=0.2, size=3,  col=c(rep("gray",5),rep("black",4),rep("blue",4), rep("red",4), rep("yellow",4),rep("violet",4),rep("brown",4),rep("blue",4),))+
# #  theme_classic()
# #geom_text( label = label_name, nudge_x = 0.1, nudge_y = 0.1,  check_overlap = T)+
# #theme_bw() + 
# #  theme(panel.border = element_blank(), panel.grid.major = element_blank(),panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"))
# #Gcol<-c(rep("gray",5),rep("black",4),rep("blue",4), rep("red",4),rep("brown",4),rep("yellow",4),rep("navy blue",3), rep("purple",4),rep("LimeGreen",4),rep("pink",4),rep("orange",4), rep("magenta",4))
# pdf("PCAanalyses_Disease2.pdf", width = 6, height = 6)
# ggplot(toplot, aes(Dim1, Dim2))+ geom_point(col="red") +
#   geom_text_repel(label=colnames(Patients),segment.size=0.01,hjust=0.2, size=3,col="red")+
#   theme_classic()+
#   #geom_text( label = label_name, nudge_x = 0.1, nudge_y = 0.1,  check_overlap = T)+
#   theme_bw() + 
#   theme(panel.border = element_blank(), panel.grid.major = element_blank(),panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"))
# 
# dev.off()
# ### 
# 
# 
# 
# 
# # reorder the column of genelist.filted
# genelist.filted<- genelist.filted[,c(2,9,10,6,8,13,3,14,11,4,15,12,1,5,18,16,7,19,17,20,21:40)]
# 
# 
# ## add the p value, p adjusted value, average of case, average of control, and logFC
# ## using the t test
# genelist.filted$pvalue=apply(genelist.filted, 1, function(x) t.test(x[1:20],x[21:40])$p.value)
# ## using the wilcoxon test
# genelist.filted$wilpvalue=apply(genelist.filted, 1, function(x) wilcox.test(x[1:20],x[21:40])$p.value)
# #genelist.filted$wilpvalue
# genelist.filted$CaseMean=apply(genelist.filted[1:20], 1, mean)
# genelist.filted$ControlMean=apply(genelist.filted[21:40], 1, mean)
# genelist.filted$Padjust= p.adjust(genelist.filted$wilpvalue, method="fdr")
# 
# genelist.filted$GFRHighCaseMean=apply(genelist.filted[11:20], 1, mean)
# #genelist.filted$ControlMean=apply(genelist.filted[21:40], 1, mean)
# 
# genelist.filted$LogFC= apply (genelist.filted, 1, function(x) log2(mean(x[1:20])/mean(x[21:40])))
# #install.packages("Metrics")
# library(pROC)
# genelist.filted$AUC = apply(genelist.filted, 1, function(x) auc(c(rep(1,20),rep(0,20)),x[1:40]))
# genelist.filted$AUC
# c(rep(1,20),rep(0,20))
# #install.packages("pROC")
# #library(pROC)
# #df_train <- data.frame( x= c(1,2,3,4,5),
# #                        y= c(1,5,8,15,26),
# #                         z=c(0,1,1,0,0))
# # df_test <- data.frame( x= c(6,7,8),
# #                        y= c(38,45,72),
# #                        z=c(0,1,0))
# # df_train
# # df_test
# # # fit logistic model
# # model <- glm(z ~ x+y, data=df_train)
# # prediction <- predict(model, df_test, 
# #                       type="response")
# # create roc curve
# 
# ### print the table our 
# write.table(genelist.filted, file="FilteredProteomic_pvalue_wilcoxon3.txt", sep="\t", quote=F)
# write.csv(genelist.filted, file="FilteredProteomic_pvalue_wilcoxon3.csv")
# 
# ### print the table with raw dataset
# row.names(genelist.filted)
# #genelist.filted
# #ProteomicMatrix
# #intersect(as.data.frame(genelist.filted), as.data.frame(ProteomicMatrix))
# 
# ## inersect the row A and row B
# 
# genelist.filted$name<-row.names(genelist.filted)
# ProteomicMatrix$name<-row.names(ProteomicMatrix)
# 
# length(intersect(genelist.filted$name, ProteomicMatrix$name))
# 
# OrginalWithInformation<- merge(genelist.filted,ProteomicMatrix, by="name")
# 
# nrow(OrginalWithInformation)
# 
# ### with all the information in details
# 
# OrginalWithInformation[is.na(OrginalWithInformation)] <- 0
# 
# 
# 
# ## print the table with the details
# 
# write.table(OrginalWithInformation, file="FilteredProteomic_pvalue_wilcoxon_WithOrginal2.txt", sep="\t", quote=F)
# 
# ### draw a viocano plot to show differentially expressed genes
# 
# # here we update the text label
# 
# ### Detect the differntiation expressed genes 
# Upregulated<- genelist.filted[genelist.filted$Padjust <= 0.1 &  genelist.filted$LogFC >=0.58, ]
# Downregulated<- genelist.filted[genelist.filted$Padjust <= 0.1 &  genelist.filted$LogFC <=-0.58, ]
# 
# nrow(Upregulated)
# nrow(Downregulated)
# 
# ### Data prepare for volcano plot
# rownames(genelist.filted)
# # add the significant details
# #volcano_data <- data.frame(GeneID=rownames(genelist.filted), Log2FoldChange=genelist.filted$LogFC, PadjustValue=genelist.filted$Padjust)
# #genelist.filted$Significant <- as.factor(ifelse(genelist.filted$Padjust <= 0.1 & abs(genelist.filted$LogFC) >= 0.58, ifelse(genelist.filted$LogFC > 0, "Up_regulated", "Down_regulated"), "Not_significant"))
# 
# # another way
# Volcano_data <- genelist.filted %>% mutate(Significant= case_when(Padjust <=0.1 & LogFC >=0.58 ~ "Up_regulated",
#                                                  Padjust <=0.1 & LogFC <= -0.58 ~ "Down_regulated",
#                               .default = "No_signficant")) %>% dplyr::select(name,LogFC,Padjust,Significant) %>% as.data.frame()
# 
# Volcano_data
# 
# # top 1 genes: 5 up-regulated and 5 down-regulated by P adjust value
# # top 10 genes: 5 up-regulated and 5 down-regulated by P-value
# top_genes <- Volcano_data %>%
#   filter(Significant != "No_signficant") %>%
#   arrange(Padjust) %>%
#   group_by(Significant) %>%
#   slice_head(n = 5) %>% 
#  ungroup()
# 
# top_genes
# Volcano_data
# 
# library(ggrepel)
# pdf("Viocanoanalyses_wilcoxon_v0515.pdf", width = 4, height = 4)
# 
# ggplot(Volcano_data,aes(x= Volcano_data$LogFC,y=-log10(Volcano_data$Padjust), color= Volcano_data$Significant))+
#   geom_point(size=0.5,color= ifelse(Volcano_data$Significant == "Up_regulated","red",
#                                     ifelse(Volcano_data$Significant == "Down_regulated","blue","gray")))+ 
#   #guides(color = guide_legend(override.aes = list(shape = 16))) +
#   #geom_text_repel(data = top_genes,
#   #                 aes(top_genes$LogFC, -log10(top_genes$Padjust)),label = top_genes$name)+
#   theme_bw()+
#   theme(plot.background = element_blank(),
#         text = element_text(size=16),
#         panel.grid.major = element_blank(),
#         panel.grid.minor = element_blank(),
#         panel.background = element_blank())  + 
#   scale_x_continuous(limits=c(-9,9),breaks =c(-9,-6,-3,0,3,6,9)) +
#   scale_y_continuous(limits=c(0,3),breaks =c(0,1,2,3)) 
# 
# 
# #+ylim(0,15) + scale_x_continuous(limits=c(-6,6),breaks =c(-6,-4,-2,0,2,4,6))
# 
# 
# dev.off()
# 
# 
# DEGRegulatedGenes
# ## write into a table
# write.table(Upregulated,file="UpregulatedGenes.txt", sep="\t", quote=F)
# write.table(Downregulated,file="DownregulatedGenes.txt", sep="\t", quote=F)
# 
# write.table(DEGRegulatedGenes, file="DEG_padjustedvalue_wilcoxon.txt", sep="\t", quote=F)
# 
# ## generated a figure to show the differentially expressed genes
# 
# length(row.names(Downregulated))
# length(row.names(Upregulated))
# ###  annotate the ensemble different expression with the gene ID
# 
# #enrichGO(row.names(FHFD_FLFD_UrotheliumcompDDEG), OrgDb = "org.Mm.eg.db", ont='BP',pAdjustMethod = 'BH',pvalueCutoff = 1, qvalueCutoff = 1,keyType = 'SYMBOL')
# #CgoUp<-enrichGO(row.names(CUDEG), OrgDb = "org.Mm.eg.db", ont='BP',pAdjustMethod = 'BH',pvalueCutoff = 0.05, qvalueCutoff = 0.5,keyType = 'ENSEMBL')
# CgoUp<-enrichGO(row.names(Upregulated), OrgDb = "org.Hs.eg.db", ont='BP',pAdjustMethod = 'BH',pvalueCutoff = 1, qvalueCutoff = 1,keyType = 'SYMBOL')
# CgoDown<-enrichGO(row.names(Downregulated), OrgDb = "org.Hs.eg.db", ont='BP',pAdjustMethod = 'BH',pvalueCutoff = 1, qvalueCutoff = 1,keyType = 'SYMBOL')
# head(CgoDown)
# CgoUpSimply <- simplify(CgoUp)
# CgoDownSimply <- simplify(CgoDown)
# enrichMap(CgoUp)
# 
# write.table(CgoUp, file="Proteomic.UpRegulatedGenes_wilcoxon.GO.txt", sep="\t", quote=F)
# write.table(CgoDown, file="Proteomic.DownRegulatedGenes_wilcoxon.GO.txt", sep="\t", quote=F)
# 
# 
# write.table(CgoUpSimply, file="Proteomic.UpRegulatedGenes_wilcoxon_simplified.GO.txt", sep="\t", quote=F)
# write.table(CgoDownSimply, file="Proteomic.DownRegulatedGenes_wilcoxon_simplified.GO.txt", sep="\t", quote=F)
# 
# #out<- topTags(res,  adjust.method = "BH", sort.by = "PValue", n="Inf")$table
# 
# #CtrCon<-merge(out,res$fitted.values,by="row.names")
# 
# #write.table(CtrCon,file="KOInfectedVsWTInfected.AllGenes.Eliminated.ignore2.txt")
# 
# 
# ##### generated the upregulated/downregulated GO term figures.
# # for Upregulated genes
# #https://bioconductor.statistik.tu-dortmund.de/packages/3.8/bioc/vignettes/ReactomePA/inst/doc/ReactomePA.html
# head(as.data.frame(CgoUp))
# 
# 
# 
# pdf("UpRegulatedGenes_barploteGO.pdf", width = 10, height = 8)
# barplot(CgoUp, showCategory = 20,font.size = 12,label_format = 80)
# dev.off()
# 
# pdf("UpRegulatedGenes_DotplotGO.pdf", width = 10, height = 8)
# dotplot(CgoUp,showCategory=20,font.size = 12,label_format = 80)
# dev.off()
# 
# d <- GOSemSim::godata("org.Hs.eg.db", ont = "BP")   
# compare_cluster_GO_emap_2 <- enrichplot::pairwise_termsim(CgoUp, semData = d,  method="Wang")
# 
# pdf("UpnRegulatedGenes_emapplot.pdf", width = 10, height = 8)
# emapplot(compare_cluster_GO_emap_2)
# dev.off()
# 
# 
# 
# #cnetplot(BgoUp, categorySize="pvalue", foldChange=geneList)
# 
# # for downregulated genes
# 
# pdf("DownRegulatedGenes_barploteGO.pdf", width = 10, height = 8)
# barplot(CgoDown, showCategory = 20,font.size = 12,label_format = 80)
# dev.off()
# 
# pdf("DownRegulatedGenes_DotplotGO.pdf", width = 10, height = 8)
# dotplot(CgoDown,showCategory=20,font.size = 12,label_format = 80)
# dev.off()
# 
# d <- GOSemSim::godata("org.Hs.eg.db", ont = "BP")   
# pdf("UDownRegulatedGenes_emapplot.pdf", width = 10, height = 8)
# compare_cluster_GO_emap <- enrichplot::pairwise_termsim(CgoDown, semData = d,  method="Wang")
# emapplot(compare_cluster_GO_emap)
# dev.off()
# 
# ## find the EGF gene to generate a expression profile
# 
# EGF<-as.data.frame(t(genelist.filted["EGF", 1:40]))
# 
# 
# 
# # add the group information
# EGF$group<-as.factor(c(rep("Case",20),rep("Control",20)))
# 
# EGF$EGF<-as.numeric(EGF$EGF)
# as.numeric(EGF$EGF)
# 
# levels(EGF$group)
# 
# EGF %>% group_by(group) %>% summarise (avg=mean(EGF))
# 
# ## generate a barplot 
# 
# pdf("EGF_Example_Expression.pdf", width = 3, height = 5)
# ggplot(data=EGF, aes(x=group, y=EGF)) + geom_boxplot(fill=c("red","gray")) +geom_jitter(shape=16, position=position_jitter(0.2))+theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(),
#                                                                                                                                                     panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"))
# wilcox.test(EGF$EGF~EGF$group)
# 
# dev.off()
# 
# 
# ### Here is to correlated the proteomic dataset to the GFR clinical factors
